Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.ts

```diff
@@ -0,0 +1,352 @@
+    if (isIconUpdateEnabled)
+      ArgumentUtility.CheckNotNullAndTypeIsObject('iconContext', iconContext);
+    ArgumentUtility.CheckTypeIsObject('commandInfo', commandInfo);
+    ArgumentUtility.CheckNotNullAndTypeIsObject('resources', resources);
+
+    this._itemBackUp = null;
+    this._isInvalidated = false;
+    BackupItemData(this, hiddenField.val(), textbox.val());
+    this._command = command;
+    this._commandBackUp = command;
+    this._selectListID = baseID + '_Results';
+    this._informationPopUpID = baseID + '_Information';
+
+    if (BrowserUtility.GetIEVersion() > 0)
+    {
+      // For Internet Explorer + JAWS, we must use the ARIA 1.0 combobox pattern.
+      // The remaining browsers support ARIA 1.1 with NVDA.
+
+      var internetExplorerScreenReaderLabelID = baseID + '_InternetExplorerScreenReaderLabel';
+      var screenReaderLabel = $("<span hidden='hidden' />")
+        .attr("id", internetExplorerScreenReaderLabelID)
+        .html(resources.InternetExplorerScreenReaderLabelText)
+        .appendTo(combobox);
+
+      textbox.attr('role', combobox.attr('role'));
+      combobox.removeAttr('role');
+
+      textbox.attr('aria-haspopup', combobox.attr('aria-haspopup'));
+      combobox.removeAttr('aria-haspopup');
+
+      textbox.attr('aria-expanded', combobox.attr('aria-expanded'));
+      combobox.removeAttr('aria-expanded');
+
+      if (combobox.attr('aria-labelledby') !== undefined)
+      {
+        textbox.attr('aria-labelledby', combobox.attr('aria-labelledby') + ' ' + internetExplorerScreenReaderLabelID);
+        combobox.removeAttr('aria-labelledby');
+
+        if (combobox.attr('data-label-id-index') !== undefined)
+        {
+          textbox.attr('data-label-id-index', combobox.attr('data-label-id-index'));
+          combobox.removeAttr('data-label-id-index');
+        }
+      }
+      else
+      {
+        textbox.attr('aria-labelledby', internetExplorerScreenReaderLabelID);
+      }
+
+      if (combobox.attr('aria-describedby') !== undefined)
+      {
+        textbox.attr('aria-describedby', combobox.attr('aria-describedby'));
+        combobox.removeAttr('aria-describedby');
+      }
+
+      combobox = textbox;
+    }
+
+    textbox.autocomplete(controlServiceUrl, 'Search', 'SearchExact',
+      {
+        extraParams: searchContext,
+        isAutoPostBackEnabled: isAutoPostBackEnabled,
+        nullValue: nullValueString, // the hidden field value indicating that no value has been selected
+        searchStringValidationParams:
+        {
+          inputRegex: new RegExp(searchStringValidationInfo.ValidSearchStringRegex),
+          dropDownTriggerRegex: new RegExp(searchStringValidationInfo.ValidSearchStringForDropDownRegex),
+          dropDownTriggerRegexFailedMessage: searchStringValidationInfo.SearchStringForDropDownDoesNotMatchRegexMessage,
+          getDropDownSearchString: function (this: BocAutoCompleteReferenceValue, searchString: string)
```

Comment:
Be careful when adding a `this: xxx` parameter to a function. TypeScript cannot validate if the correct `this` is provided. In this case, `this` will not be what you expect it to be. Instead a lambda expression should be used to capture the current `this` context.